src/extlibs/hrvo/simulator.cpp

```diff
@@ -134,9 +134,10 @@ void Simulator::updatePrimitiveSet(const TbotsProto::PrimitiveSet &primitive_set
                 {
                     // TODO (#2418): Update implementation of Primitive to support
                     // multiple path points
+                    auto destination = primitive.move().path().point().at(0);
                     goal->positions_.emplace_back(
-                        static_cast<float>(primitive.move().destination().x_meters()),
-                        static_cast<float>(primitive.move().destination().y_meters()));
+                        static_cast<float>(destination.x_meters()),
+                        static_cast<float>(destination.y_meters()));
```

Comment:
`Goal` in HRVO actually supports multiple points in a path, so we should ideally add all points in the path here. 
Something else that I think I missed when updating the destination is updating `Goal`'s `speedAtPosition_` (which is an array of the same length as the `positions_` array with speeds of the path point with the same index). That would also be nice if we can update similar to how we are updating `goal->positions_`.

Reply:
I also made a few changes here with the new `Path` instead of `Goal` object in my PR. It should be scalable if we want to also update with `speedAtPosition` for each `PathPoint`.